CDTDatastore/CDTReplicator/CDTReplicator.m

```diff
@@ -288,6 +314,75 @@ - (BOOL)startWithTaskGroup:(dispatch_group_t)taskGroup error:(NSError *__autorel
     return YES;
 }
 
+/**
+ Builds a TDReplicator object from a CDTPush/PullReplication configuration.
+ */
+- (TDReplicator *)buildTDReplicatorFromConfiguration:(CDTAbstractReplication *)configuration
+                                               error:(NSError *__autoreleasing *)error
+{
+    // This validates the optional headers
+    NSMutableDictionary *properties = [[NSMutableDictionary alloc] init];
+    
+    if ([CDTAbstractReplication validateOptionalHeaders:configuration.optionalHeaders error:error]) {
```

Comment:
`validateOptionalHeaders` is not needed here since it was already called by `initWithTDDatabaseManager`